### DISCUSSION TITLE
How to use multi-value option or all option in a variable ?

### DISCUSSION THREAD

**Discussion by philippesg** [grafana/grafana-json-datasource#273] [Q&A]
Hello, perhaps, it's a easy but i don't find how to use the multi-value option or 'all" option in a variable with json api. Could you help me how to make multiselection applicable for filter values which is consuming json api ?

i using multi-value in my 1st variable , and it displayed my values 
![image](https://user-images.githubusercontent.com/49228865/162200643-4d1cbdd0-467b-4a75-9943-01b94896eecf.png)

But, the second don't display any values because i don't know how to use 'team variable' in the second variable :(  

the sign == dont work i don't know what i can replace it.

 $.[?(@.Teams == '$teams')].SIP_service_account_used 

![image](https://user-images.githubusercontent.com/49228865/162200086-be98a79e-373c-4e2a-ac24-7f3766fb68bc.png)

Have you an idea ? Could you help me ?

**Comment by marcusolsson** [comment]
The problem here is that `$teams` will be expanded to something like `@.Teams == 'team1, team2, team3'`. To support this, we'd need to add a custom formatter, e.g. `${teams:jsonpath}`, that would expand to something like this:

```js
@.Teams == 'team1' || @.Teams == 'team2' || @.Teams == 'team3'
```

Would that solve your problem?

**Reply by philippesg** [reply]
Hello, Marcusolsson, thanks you for your reply. I was on holiday last week, i'm seeing your answer just now. 

I'm getting an error when i'm trying the custom formatter : 

`$.[?(@.Teams in ${teams:jsonpath})].SIP_service_account_used`

**jsonPath: invalid or unexpected token:_$_v.Teams in CCS
Error updating options : jsonPath: invalid or unexpected toker:_$_v.Teams in CC**S

I don't understand this error, could you help me ?

**Reply by marcusolsson** [reply]
Yes, it's currently not supported by the plugin. In my reply I was exploring options on how to implement it.

**Comment by philippesg** [comment]
Do you know if it is possible to have an update of the plugin to support it ?

**Reply by marcusolsson** [reply]
I don't know when I'll have time to add it myself, but I'd be happy to accept a pull request!
